src/macaron/slsa_analyzer/checks/build_as_code_check.py

```diff
@@ -366,19 +366,22 @@ def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResu
         ci_services = ctx.dynamic_data["ci_services"]
 
         # Check if "build as code" holds for each build tool.
+        overall_res = CheckResultType.FAILED
+
         for tool in build_tools:
             res = self._check_build_tool(tool, ctx, ci_services, check_result)
```

Comment:
I think there might be issues if we run `self._check_build_tool` multiple times because at this [line](https://github.com/oracle/macaron/blob/68a54ac5573460e8d61fcb9662baa81c783c0c09/src/macaron/slsa_analyzer/checks/build_as_code_check.py#L224), we are creating a new list with the corresponding `BuildAsCodeFacts` table (which will be stored into the database). 
The list was created this way because we have been assuming to stop on a passing BuildAsCode run for a single build tool. 
Now if we run it multiple times, this list will be overwritten each time :thinking: , whereas the expected behavior is that each time we run `self._check_build_tool` with a new build tool, a different `BuildAsCodeFacts` instance should be appended to the list. Can you have a look to see if this is the case in terms of the values stored into the database? The `BuildAsCodeFacts` instance will be stored as the table called `_build_as_code_check`.

Reply:
Thanks for the catch, far as I can see it was .append() in my original PR (but also there was still an equals in one spot either because I missed it or some other reason?), but it changed to an equals at some point. In any case I've updated it now so it should hopefully be all good. I've also made a couple tests for this in my latest commit, do they look alright?